libsg3/ses/ses.pyx

```diff
@@ -329,11 +329,12 @@ cdef class EnclosureDevice(object):
         }
         return element_dict[index]
 
-    def decode_response(self, buff):
+    cdef str decode_response(self, char * buff):
```

Comment:
Wait, since this can raise exception you need to simplify this function and do it the cython way
```
cdef str decode_request(self, char * buff) except -1:
    cdef bytes py_bytes = buff
    return py_bytes.decode(errors='replace')

Reply:
According to docs [here](https://cython.readthedocs.io/en/latest/src/userguide/language_basics.html#id15), we should not declare exception values for functions returning Python objects. If we were returning C int/float/char, in that case it is needed.

I also get an error during compile time if I do this:
```
Exception clause not allowed for function returning Python object
```

Reply:
Either way you don't need to do the `try: catch: raise`. You can just do `return py_bytes.decode(errors='replace')` and if that crashes it'll bubble up all the same.

Reply:
Thanks, I have removed the `try: catch: raise`.